src/number/parseHumanDuration.ts

```diff
@@ -0,0 +1,114 @@
+export type HumanQuantity<
+  Unit extends string,
+  ShortUnit extends string = never,
+> = `1 ${Unit}` | `${number} ${Unit}s` | `${number}${ShortUnit}`
+
+export type HumanQuantityOptions<
+  Unit extends string = string,
+  ShortUnit extends string = string,
+> = {
+  /**
+   * Time units mapping
+   */
+  units: Record<Unit, number>
+  /**
+   * Aliases for time units
+   */
+  short?: Record<ShortUnit, Unit>
+}
+
+/**
+ * Defines a parser for human quantity strings
+ *
+ * @see https://radashi.js.org/reference/number/parseHumanDuration
+  * @example
+  * ```ts
+  * const distanceParser = _.defineHumanQuantityParser({
+  *   units: {
+  *     kilometer: 1_000,
+  *     mile: 1_852,
+  *     yard: 0.9144,
+  *     foot: 0.3048,
+  *     meter: 1,
+  *   },
+  *   short: {
+  *     km: 'kilometer',
+  *     mi: 'mile',
+  *     yd: 'yard',
+  *     ft: 'foot',
+  *     m: 'meter',
+  *   },
+  * })
+  *
+  * distanceParser("1 kilometer") // => 1_000
+  * distanceParser("1 mile") // => 1_852
+  * distanceParser("1 yard") // => 0.9144
+  * distanceParser("1ft") // => 0.3048
+  * distanceParser("1 meter") // => 1
+  * ```
+  */
+export function defineHumanQuantityParser<
+  Unit extends string,
+  ShortUnit extends string = never,
+>({ units, short }: HumanQuantityOptions<Unit, ShortUnit>) {
+
+  return (quantity: HumanQuantity<Unit, ShortUnit>): number => {
+    const match = quantity.match(/^(-?\d+(?:\.\d+)?) ?(\w+)?s?$/)
+    if (!match) throw new Error(`Invalid quantity, cannot parse: ${quantity}`)
+
+    let unit = match[2] as Unit | ShortUnit
+    unit = short && unit in short ? short[unit as ShortUnit] : (unit as Unit)
+
+    let count = Number.parseFloat(match[1])
```

Comment:
I saw no reason for this to be a let

```suggestion
    const count = Number.parseFloat(match[1])
```